src/views/RecIM/components/Forms/Form/index.jsx

```diff
@@ -0,0 +1,214 @@
+    return matchingField.label;
+  };
+
+  function getNewFields(currentInfo, newInfo) {
+    const updatedFields = [];
+    Object.entries(newInfo).forEach(([key, value]) => {
+      if (currentInfo[key] !== value)
+        updatedFields.push({
+          Field: key,
+          Value: value,
+          Label: getFieldLabel(key),
+        });
+    });
+    return updatedFields;
+  }
+
+  const handleWindowClose = () => {
+    setOpenConfirmWindow(false);
+    setOpenForm(false);
+    setNewInfo(currentInfo);
+  };
+
+  /**
+   * @param {Array<{name: string, label: string, type: string, menuItems: string[]}>} fields array of objects defining the properties of the input field
+   * @returns JSX correct input for each field based on type
+   */
+  const mapFieldsToInputs = (fields) => {
+    return fields.map((field) => (
+      <InformationField
+        key={field.name}
+        error={field.error}
+        label={field.label}
+        name={field.name}
+        helperText={field.helperText}
+        value={newInfo[field.name]}
+        type={field.type}
+        menuItems={field.menuItems}
+        onChange={handleChange}
+        xs={12}
+        sm={6}
+        md={4}
+        lg={3}
+      />
+    ));
+  };
+
+  const DialogContent = () => {
+    let index = 0;
+    return fields.map((set) => (
+      <ContentCard
+        title={
+          formTitles.contentCardTitles
+            ? formTitles.contentCardTitles[index++]
+            : `${formTitles.name} Information`
+        }
+      >
+        {additionalContent}
+        {loading ? <GordonLoader /> : mapFieldsToInputs(set)}
+      </ContentCard>
+    ));
+  };
+
+  return (
+    <GordonDialogBox
+      open={openForm}
+      title={`${formTitles.formType} ${formTitles.name}`}
+      fullWidth
+      maxWidth="lg"
+      buttonClicked={() => {
+        // This will submit the data and close the window if we do not allow open confirm
+        if (showConfirmationWindow) {
+          setOpenConfirmWindow(true);
+        } else {
+          handleConfirm(handleWindowClose);
+        }
+      }}
+      isButtonDisabled={disableUpdateButton}
+      buttonName={'Submit'}
+      cancelButtonClicked={() => {
+        setNewInfo(currentInfo);
+        setOpenForm(false);
+        if (additionCancelActions) {
+          additionCancelActions();
+        }
+      }}
+      cancelButtonName={'cancel'}
+    >
+      {DialogContent()}
+
+      {!loading && (
+        <GordonDialogBox
+          open={openConfirmWindow}
+          title={`Confirm Your ${formTitles.name}`}
+          buttonClicked={() => {
+            !isSaving && handleConfirm(newInfo, handleWindowClose);
+          }}
+          buttonName="Confirm"
+          // in case you want to authenticate something change isButtonDisabled
+          isButtonDisabled={false}
+          cancelButtonClicked={() => {
+            if (!isSaving) setOpenConfirmWindow(false);
+          }}
+          cancelButtonName="Cancel"
+        >
```

Comment:
This is more of a big picture thing, but I think the nested dialog here is really unfortunate. I would prefer a two-step form within a single dialog (i.e., you fill out the form and then hit a `Next` button, which shows you the confirmation step and the Submit button.

Also, for most of these I don't think you need the `Confirm` step at all. My rule of thumb would be, if the user can easily correct mistakes (e.g. fixing a team name) then the confirm is more cumbersome than useful.